modules/twinklewarn.js

```diff
@@ -42,9 +42,22 @@ Twinkle.warn = function twinklewarn() {
 	} else if (mw.config.get('wgDiffNewId') || mw.config.get('wgDiffOldId')) {
 		// Autofill user talk links on diffs with vanarticle for easy
 		// warning, but don't autowarn
-		var warnFromTalk = function(talkLink) {
+		var warnFromTalk = function(xtitle) {
+			var talkLink = $('#mw-diff-' + xtitle + '2 .mw-usertoollinks a').first();
 			if (talkLink.length) {
 				var extraParams = 'vanarticle=' + mw.util.rawurlencode(Morebits.pageNameNorm) + '&' + 'noautowarn=true';
+
+				// This relies upon the text shown above a diff, which depends upon the user's
+				// language selection and date format preference. While it could thus vary widely,
+				// once we remove the leading text, the date formats remain the same (fails of RTL).
+				// Of the four date formats users can select, only one is a
+				// valid Date format; the other three require removal of a comma.
+				var dateText = $('#mw-diff-' + xtitle + '1 a').first().text().replace(/^.*?(\d\d)/, '$1');
+				var vantimestamp = new Date(dateText.replace(/^(\d\d:\d\d),/, '$1')).getTime();
```

Comment:
Fails for RTL? This will fail on any language other than English, since JS Date() can't comprehend non-English month names. There are also languages that use non-Arabic numerals (such as Hindi) which aren't recognised by `\d`. 

This should be written as: 
```js
var diffTitleText = $('#mw-diff-' + xtitle + '1 a').first().text();
var dateobj = new Date(diffTitleText.replace(/^.*?(\d\d)/, '$1').replace(/^(\d\d:\d\d),/, '$1')); // language specific
var vantimestamp = dateobj.getTime();
```
so that the part that needs changing b/w wikis is all together.

Reply:
Thanks, I dunno what I was smokin' when I wrote that.

In reality, this segment (hijacking the user talk links) isn't really tenable without either adding something like `wgDiffOldTimestamp` or an API call.  Relying on parsing text that is localized and formatted according to user preferences and customizable by wiki is just setting up for failure.  I think the former would be useful but I'd imagine the devs wouldn't be likely to add it.

Reply:
Continued below